raytracing/rays.py

```diff
@@ -284,18 +286,14 @@ def __init__(self, yMax=1.0, yMin=None, thetaMax=pi / 2, thetaMin=None, maxCount
         self.thetaMin = thetaMin
         if thetaMin is None:
             self.thetaMin = -thetaMax
-        super(RandomRays, self).__init__(rays=None)
+        super(RandomRays, self).__init__()  # Even with this, rays = []
+        self.rays = None
 
     def __len__(self) -> int:
         return self.maxCount
 
     def __getitem__(self, item):
-        if self.rays is None:
-            raise NotImplementedError("You cannot access RandomRays")
-        elif len(self.rays) < item:
-            raise NotImplementedError("You cannot access RandomRays")
-        else:
-            return self.rays[item]
+        raise NotImplementedError("You cannot access RandomRays")
 
     def __next__(self) -> Ray:
         if self.iteration >= self.maxCount:
```

Comment:
This will need to be modified too to make sure the behaviour is correct: generate next if not already there, if not, use self.rays[self.iteration]